tempesta_fw/t/functional/helpers/control.py

```diff
@@ -320,23 +395,28 @@ def letout_finish(self):
         if self.module_stap:
             self.node.run_cmd('rm -r %s' % self.modules_dir)
 
-    def start(self):
-        Tempesta.start(self)
+    def run_start(self):
+        Tempesta.run_start(self)
         self.inject_prepare()
         self.inject()
 
-    def stop(self):
-        self.letout()
-        self.letout_finish()
-        Tempesta.stop(self)
+    def stop_tempesta(self):
+        letout_fail = None
+        try:
+            self.letout()
+            self.letout_finish()
+        except Exception as exc:
+            letout_fail = exc.message
+        Tempesta.stop_tempesta(self)
+        if letout_fail != None:
+            raise Exception("letout finish failed: %s" % letout_fail)
```

Comment:
The whole patch was about not throwing exceptions during stop process to let all classes make theirs best to cleanup environment. This line breaks all your efforts. Take your time, don't hurry.

Reply:
fixed